server/app/jobs/node_cleanup_job.rb

```diff
@@ -25,7 +25,8 @@ def perform
   end
 
   def cleanup_stale_connections
-    HostNode.where(:last_seen_at.lt => NODE_DISCONNECT_TIMEOUT.ago).each do |node|
+    HostNode.where(:connected => true, :last_seen_at.lt => NODE_DISCONNECT_TIMEOUT.ago).each do |node|
```

Comment:
Is this `HostNode.where(:connected => true)` legit? Before, it was re-running this loop on all disconnected nodes every minute.